### ISSUE TITLE
Character limit on 'Title', 'Presenter' & 'Subtitle'

### ISSUE THREAD

**Issue by djelton92** [cpvalente/ontime#145]
This software has huge potential, so we already started to event-test it a bit when we ran into a tiny-but-big issue:

'Title', 'Presenter' & 'Subtitle' fields all have too small character limit. For example when the presentations name is longer, the moderator can't introduce the actual title, but the summarised title. And most importantly, a lot of online events have multi-panel sessions where you have up to 8/9 people in the same session.
Testing it in developer mode it seems like it should work and not interfere with something, but I'm not code-smart enough to remove the char limit myself.

Would it be an "easy fix" or something more demanding?

**Comment by cpvalente** [comment]
Hey @djelton92, thank you for your comment. happy to hear that you are testing the app and happy that you are getting involved with feature requests.

Just had a quick look and it seems that the character limit is set at 40 characters. 
At first, this was set in collaboration with a client on a size which we considered was the minimum for displaying a clear message at a certain size screens in a stage environment (cant really remember the specifics).

Bottom line is: character limit exists to ensure the viewers have readable content.
I can of course revise the character limit, but this would mean revisiting the designs of the viewers. 

How is this behaviour implemented in other applications? What would a reasonable cap be (if any at all)?

**Comment by djelton92** [comment]
Hey!

Yes, 40 characters is way too little, but I understand the limitations.
Is it possible to just remove it (or show/tell me how to remove it only in my computer)?

I understand the reasoning, but for example the text to the moderator isn't character limited and also goes over the edge if you input too much. So I think the first step would be to remove the limitations - you type as much as it seems to fit.

And second step would be to make display modes scalable. So the text wouldn't be a certain px size, but relative to the whole UI. It may be possible to set it for example instead of font=12px, it could maybe be like font=0.5vh (viewheight).


To be honest, we have done over a few hundred events over the past years with just Excel/Spreadsheet. Starting the timers manually in vmix and using different views inside vmix to combine the views you're offering. This is why I'm pushing this solution in our company right now.


I'd honestly help, but my skills are limited to a bit of php, html, css. So maybe I can help and try to figure out the text scaling on some of the view modes?


Thank you for your fast responses, hope to use this in our next production next week ;)

**Comment by cpvalente** [comment]
Hey @djelton92

### Character limiting
The UI app is responsible for the validation and character limiting as of now. 
Few options to make it work:
- You could need the maxchar in EditableText event to something like 999 (ExpandedBlock.jsx lines 39,59,65,71 and CollapsedBlock.jsx line 44. This means building the app yourself.
- The parsing functions dont have the same limitation so you could edit the db file by hand or import from excel. either of this should go arount the limit

I am working in a redesign for the Event component so it is unlikely that I will push any changes towards this in the next week, but if you get in touch at mail@getontime.no I am happy to send you a release with the restrictions lifted

### Responsive views and some help
Better responsive views are definitely something that would improve the app and I am thankful if you are willing to help.
Html and CSS skills is all you need since the data layers are already build.
Few challenges:

The text is already responsive in the way that you describe
title = font-size: 2.5vw;
subtitle =  `font-size: 1.5vw;`
presenter = ` font-size: 2vw;`

but it does not attempt resizing the content depending on its length or allowing text to overflow
```
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
```
it is also important noting that on the cards that show current and next event (inside the Timer view) the width is limited to 47.5vw on each

Again, feel free to get in touch and we can look into how this could proceed

**Comment by je71175** [comment]
Just to add our voice the the request for more characters also in the cuesheet custom fields too please.  Sometimes a single cell has a lot of data which needs to be captured and displayed.

**Comment by emmdubb** [comment]
Heya @cpvalente, following up on my reddit post from yesterday.

There are two aspects to this issue.   

The data entry side that is currently preventing us from saving strings longer than 40 characters, and also prevents us from pasting too long strings.  But it sounds like you're already addressing this.

On the display side for the various views, I've got a couple of different ideas how long strings can be handled.

The first option is to simply truncate the string, but have the full string appear on mouseover with a tooltip.

The second option would be to implement a collapse/expand mechanism, similar to how you're already dealing with events in the Events List.  In it's collapsed view, we see the truncated information.  When we expand the line it expands to show the whole text.

I'm not sure if this is helpful or not, but back in the day when I was still doing web development, I would do math with js to see how many characters I can fit in the element, truncate accordingly and dump the full string into a data attrib.  Add an onmouseover event to tweak display as necessary, either with a modal popup or resize the element responsively and update the content.  But this was years ago, not sure how relevant that is to how Ontime is built.

Cheers!

-m

**Comment by cpvalente** [comment]
Done in v2
